qiskit/circuit/add_control.py

```diff
@@ -69,13 +69,20 @@ def _control_predefined(operation, num_ctrl_qubits):
     elif operation.name == 'h':
         import qiskit.extensions.standard.ch
         cgate = qiskit.extensions.standard.ch.CHGate()
-    elif operation.name == 'rz':
-        import qiskit.extensions.standard.crz
-        cgate = qiskit.extensions.standard.crz.CrzGate(*operation.params)
+    elif operation.name in {'rx', 'ry', 'rz'}:
```

Comment:
I don't believe a special case needs to be made for rx and ry, like it does for rz, since I believe they have a consistent phase definition with respect to their controlled counterparts.

Reply:
Hi, thanks for your comment.
I'm not entirely sure I understand your comment. Do you mean something like this?
```
elif operation.name == 'rx':
     ...
elif operation.name == 'ry':
     ...
elif operation.name == 'rz':
     ...
```
Because in that case, we did the additional case distinction to reduce the number of return statements. Otherwise Lint will raise an error because of too many return statements in one function.

Reply:
Ok, let's leave it as is for now in that case and address it later, perhaps with a dictionary. Also, it's probably the case that what's done for the rotation gates, using the definition of Cr<x>, instead of R<x>, for num_ctrl_qubits >=2, is probably better than using the base gate for num_ctrl_qubits >= 1 as done in other cases. If that's true it probably should also be applied to other gates e.g. u1, h, etc.